swift/StableDiffusion/pipeline/CGImage+vImage.swift

```diff
@@ -130,3 +130,18 @@ extension CGImage {
     }
 }
 
+extension vImage_Buffer {
+    func unpaddedData() -> Data {
+        let bytesPerPixel = self.rowBytes / Int(self.width)
+        let bytesPerRow = Int(self.width) * bytesPerPixel
+
+        var contiguousPixelData = Data(capacity: bytesPerRow * Int(self.height))
+        for row in 0..<Int(self.height) {
+            let rowStart = self.data!.advanced(by: row * self.rowBytes)
+            let rowData = Data(bytes: rowStart, count: bytesPerRow)
+            contiguousPixelData.append(rowData)
+        }
```

Comment:
I couldn't think of a way to avoid the loop, since each row contains padding at the end. I haven't profiled the code yet, but we could potentially improve performance, if necessary, using these alternatives:

- Short-circuit the loop and use a fast path if no padding has been applied.
- Use a chunk of memory and `memcpy` instead of appending `Data`.

Reply:
@alejandro-isaza Any ideas for avoiding the loop? 🙏

Reply:
Using `[UInt8]` can be a lot faster than using Data. Other than that, the loop will be optimized by the compiler, I bet it's as fast as it can get.